Engine/Main/GameObject.cpp

```diff
@@ -355,6 +357,14 @@ ENGINE_API Component* GameObject::CreateComponent(const Component::ComponentType
 	return created_component;
 }
 
+
+ENGINE_API Component* GameObject::CreateComponentUI(const ComponentUI::UIType ui_type)
+{
+	Component *created_component = App->ui->CreateComponentUI(ui_type, this);
+	components.push_back(created_component);
+	return created_component;
+}
```

Comment:
This should be inside `CreateComponent` method using different `ComponentType` definitions. Reasons for this:
- `App->ui->CreateComponentUI` has a different logic from `CreateComponent` method. Also, its not setting the owner GameObject of  the created Component. 
- Everytime that we modify the logic of creating a component we have to remember to modify this function as well.